### ISSUE TITLE
Automatic double-quote insertion broken

### ISSUE THREAD

**Issue by jbclements** [racket/drracket#351]
In the current git master, it appears that insertion of double-quotes is broken when inserting
double-quotes before other double-quotes.

How to reproduce:
1) Start DrRacket, select the "Automatic Parentheses & Quotes" option in editing>general.
2) Open a new buffer
3) hit return three times, type a double-quote. See two double-quotes inserted, as expected.
4) hit the up arrow twice. Type a double quote. See only *one* double-quote inserted, and 
syntax highlighting become angry red.

I believe this error does not affect the v7.6 release.

**Comment by rfindler** [comment]
Might be because of f7702f30d0759bcadaf8f0ccd40a7770addad2bd?

**Comment by rfindler** [comment]
Maybe because of https://github.com/racket/gui/commit/f7702f30d0759bcadaf8f0ccd40a7770addad2bd ?

**Comment by jbclements** [comment]
Yes, that seems very likely. Gosh, I figured this one was in the drracket repo for sure. Ah well. I’ll move the issue over and see if I can get bzyeung to take a look at it, if that sounds right to you.

John
